### PR TITLE
fix(observability): demote WhatsApp ingest SQLite lock noise 

### PR DESCRIPTION
**Pull request by YellowSnnowmann**
## Summary

- Added `ExpectedErrorKind::WhatsAppDataSqliteBusy` for WhatsApp ingest lock-contention errors.

- Added a narrow classifier that matches only the WhatsApp ingest envelope (`[whatsapp_data] ingest failed` + `upsert wa_message`) with SQLite lock markers.

- Wired the new error kind into expected-error reporting so this known transient condition is demoted from Sentry error events.

- Added unit tests for both positive and negative cases to protect classification boundaries.

- Kept behavior unchanged for non-WhatsApp and non-ingest SQLite lock failures to avoid over-demotion.

## Problem

- Sentry issue TAURI-RUST-BH reports repeated errors from `openhumman.whatsapp_data_ingest` with `database is locked` / `Error code 5`.

- The WhatsApp write path already has retries and busy timeout, but longer-lived external file locks can still exceed retry windows.

- These events are expected transient local-contention conditions and currently create alert noise without actionable remediation.

## Solution

- Added a new expected-error kind specific to WhatsApp ingest SQLite busy/locked failures.

- Implemented strict message matching to avoid demoting unrelated lock errors:

- Requires WhatsApp ingest failure prefix.

- Requires `upsert wa_message` context.

- Requires SQLite lock indicators (`database is locked`, `database table is locked`, `database file is locked`, or `error code 5`).

- Routed this kind through observability expected-error handling (warn breadcrumb, no Sentry error event).

- Added tests covering:

- Canonical and wrapped WhatsApp ingest lock messages.

- Negative cases from other SQLite lock domains to prevent false positives.

## Submission Checklist

If a section does not apply to this change, mark the item as N/A with a one-line reason. Do not delete items.

* Tests added or updated (happy path + at least one failure / edge case) per Testing Strategy
* Diff coverage ≥ 80% — changed lines (Vitest + cargo-llvm-cov merged via diff-cover) meet the gate enforced by `.github/workflows/coverage.yml`. Run `pnpm test:coverage` and `pnpm test:rust` locally; PRs below 80% on changed lines will not merge.
* Coverage matrix updated — added/removed/renamed feature rows in `docs/TEST-COVERAGE-MATRIX.md` reflect this change (or N/A: behaviour-only change)

N/A: behaviour-only observability classification change.

* All affected feature IDs from the matrix are listed in the PR description under ## Related

N/A: no feature-row change.

* No new external network dependencies introduced (mock backend used per Testing Strategy)
* Manual smoke checklist updated if this touches release-cut surfaces (`docs/RELEASE-MANUAL-SMOKE.md`)

N/A: no release-cut UI/runtime surface changes.

* Linked issue closed via Closes #NNN in the ## Related section

N/A: tracked via Sentry issue (no linked GitHub issue in this PR).

## Impact

- Runtime/platform: Rust core observability path only; impacts desktop telemetry classification.

- Performance: negligible string-match checks in error classification path.

- Security/privacy: reduces unnecessary Sentry error noise for known local contention cases.

- Compatibility: no API/schema/data-model changes; no migration required; no breaking changes expected.

## Related

Closes: [TAURI-RUST-BH](https://sentry.tinyhumans.ai/organizations/tinyhumans/issues/487/?project=4)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved WhatsApp data synchronization by better managing transient database lock errors to prevent unnecessary error alerts while maintaining proper logging and monitoring.

[Review Change Stack](https://app.coderabbit.ai/change-stack/tinyhumansai/openhuman/pull/2917?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/core/observability.rs [hunk 1]
```diff
@@ -183,6 +183,17 @@ pub enum ExpectedErrorKind {
...
+    /// (`SQLITE_BUSY` / `SQLITE_LOCKED`) after exhausting the local retry
+    /// budget. This is an expected local-contention condition (typically on
+    /// Windows when another process briefly holds a file lock) and the
+    /// scanner retries on the next tick, so Sentry has no immediate
+    /// remediation path.
+    ///
+    /// Anchored narrowly to the whatsapp ingest failure envelope plus the
+    /// SQLite lock text, so unrelated DB lock errors in other domains still
+    /// reach Sentry.
+    WhatsAppDataSqliteBusy,
```
[the remark this answers is not in the document]
Comment (YellowSnnowmann) [reply src/core/observability.rs:196]: In this PR I kept the change minimal but did apply the split for the newly added surface:

- moved the WhatsApp lock classifier into observability::matcher
- moved its reporting branch into observability::reporting
- moved its focused tests into observability::tests
- wired mod matcher; mod reporting; #[cfg(test)] mod tests; in observability.rs and re-exported used symbols
This keeps behavior unchanged while preventing additional growth in the monolith. A full module-wide decomposition is a larger refactor and should be done in a dedicated follow-up PR to avoid high-risk churn.

**Review by graycyrus — APPROVED** [review]
clean observability fix. the double-anchor approach (ingest envelope + upsert operation context) is the right call — narrow enough to avoid demoting unrelated lock failures in other whatsapp_data operations like list_messages.

a few things i checked:

- classifier ordering in `expected_error_kind`: the new check sits between `MemoryStoreBreakerOpen` and `DiskFull`, which is fine since every check is an independent early return with no ordering dependency
- `error code 5` as a lock indicator: risk of false positives is real in isolation, but gated behind both envelope and operation anchors so it's safe here
- `tracing::warn!` level for the reporting arm is correct — matches the pattern for other expected-error kinds, not silently swallowed
- no `.unwrap()`, no exposed private fns, no PII in the structured warn log
- positive tests cover the canonical case and the rpc-wrapped envelope; negative cases cover other domains, other operations, and a different whatsapp_data op — solid coverage of classifier boundaries

CI 100% green including coverage gate and Rust quality checks. approving.
